pkg/gds/admin_test.go

```diff
@@ -0,0 +1,939 @@
+		in: &admin.AuthRequest{
+			Credential: refresh,
+		},
+	}
+	res := s.doRequest(request, nil)
+	require.Equal(http.StatusUnauthorized, res.StatusCode)
+
+	// Invalid access token
+	request.headers = map[string]string{
+		"Authorization": "Bearer invalid",
+	}
+	res = s.doRequest(request, nil)
+	require.Equal(http.StatusUnauthorized, res.StatusCode)
+
+	// Missing refresh token
+	request.in = &admin.AuthRequest{}
+	request.headers = map[string]string{
+		"Authorization": "Bearer " + access,
+	}
+	res = s.doRequest(request, nil)
+	require.Equal(http.StatusUnauthorized, res.StatusCode)
+
+	// Invalid refresh token
+	request.in = &admin.AuthRequest{
+		Credential: "invalid",
+	}
+	res = s.doRequest(request, nil)
+	require.Equal(http.StatusUnauthorized, res.StatusCode)
+
+	// Mismatched access and refresh tokens
+	claims.Id = uuid.NewString()
+	otherToken := jwt.NewWithClaims(jwt.SigningMethodRS256, claims)
+	other, err := a.tokens.Sign(otherToken)
+	require.NoError(err)
+	request.in = &admin.AuthRequest{
+		Credential: refresh,
+	}
+	request.headers = map[string]string{
+		"Authorization": "Bearer " + other,
+	}
+	res = s.doRequest(request, nil)
+	require.Equal(http.StatusUnauthorized, res.StatusCode)
+
+	// Successful reauthentication
+	request.in = &admin.AuthRequest{
+		Credential: refresh,
+	}
+	request.headers = map[string]string{
+		"Authorization": "Bearer " + access,
+	}
+	res = s.doRequest(request, nil)
+	require.Equal(http.StatusOK, res.StatusCode)
+	// Double cookie tokens should be set
+	cookies := res.Cookies()
+	require.Len(cookies, 2)
+	for _, cookie := range cookies {
+		require.Equal(a.conf.CookieDomain, cookie.Domain)
+	}
+}
+
+// Test that the Summary endpoint returns the correct response.
+func (s *gdsTestSuite) TestSummary() {
+	require := s.Require()
+	a := initAdmin(s)
+
+	request := &apiRequest{
+		fn:     a.Summary,
+		method: http.MethodGet,
+		path:   "/v2/summary",
+	}
+	actual := &admin.SummaryReply{}
+	rep := s.doRequest(request, actual)
+	require.Equal(http.StatusOK, rep.StatusCode)
+
+	// Test against the expected response
+	expected := &admin.SummaryReply{
+		VASPsCount:           len(s.dbVASPs),
+		PendingRegistrations: 5,
+		ContactsCount:        40,
+		VerifiedContacts:     28,
+		CertificatesIssued:   0,
+		Statuses: map[string]int{
+			pb.VerificationState_APPEALED.String():       1,
+			pb.VerificationState_ERRORED.String():        1,
+			pb.VerificationState_PENDING_REVIEW.String(): 2,
+			pb.VerificationState_REJECTED.String():       2,
+			pb.VerificationState_SUBMITTED.String():      2,
+			pb.VerificationState_VERIFIED.String():       6,
+		},
+		CertReqs: map[string]int{},
+	}
+	require.Equal(expected, actual)
+}
+
+// Test that the Autocomplete endpoint returns the correct response.
+func (s *gdsTestSuite) TestAutocomplete() {
+	require := s.Require()
+	vasps := []string{
+		"d9da630e-41aa-11ec-9d29-acde48001122",
+		"d9efca14-41aa-11ec-9d29-acde48001122",
+	}
+	WriteVASPs(s, vasps)
```

Comment:
Ok, I totally see why you're trying to limit the response back, but I am ok with us not testing exact responses. I'm perfectly happy with using `require.Len` to check we get the expected number back, and maybe a spot check with `require.Contains` a few things we're expecting. We want to be a bit flexible to handle changes in the future.